contracts/compliance/ICompliance.sol

```diff
@@ -1,5 +1,83 @@
+/**
+ *     NOTICE
+ *
+ *     The T-REX software is licensed under a proprietary license or the GPL v.3.
+ *     If you choose to receive it under the GPL v.3 license, the following applies:
+ *     T-REX is a suite of smart contracts developed by Tokeny to manage and transfer financial assets on the ethereum blockchain
+ *
+ *     Copyright (C) 2019, Tokeny sàrl.
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
 pragma solidity ^0.6.0;
 
 interface ICompliance {
-    function canTransfer(address _from, address _to, uint256 value) external view returns (bool);
+
+   /**
+    *  @dev checks that the transfer is compliant.
+    *  default compliance always returns true
+    *  READ ONLY FUNCTION, this function cannot be used to increment
+    *  counters, emit events, ...
+    *  @param _from The address of the sender
+    *  @param _to The address of the receiver
+    *  @param _amount The amount of tokens involved in the transfer
+    */
+    function canTransfer(address _from, address _to, uint256 _amount) external view returns (bool);
+
+   /**
+    *  @dev function called whenever tokens are transferred
+    *  from one wallet to another
+    *  this function can update state variables in the compliance contract
+    *  these state variables being used by `canTransfer` to decide if a transfer
+    *  is compliant or not depending on the values stored in these state variables and on
+    *  the parameters of the compliance smart contract
+    *  @param _from The address of the sender
+    *  @param _to The address of the receiver
+    *  @param _amount The amount of tokens involved in the transfer
+    */
+    function transferred(address _from, address _to, uint256 _amount) external returns (bool);
```

Comment:
This function is purely informative for the compliance contract, I think it shouldn't return a boolean.

Reply:
solved here https://github.com/TokenySolutions/T-REX/commit/48f859a6eb7bf734b9cb014b0e1b74cf40453993